packages/astro/src/core/session.ts

```diff
@@ -462,15 +469,39 @@ export class AstroSession<TDriver extends SessionDriverName = any> {
 	}
 }
 // TODO: make this sync when we drop support for Node < 18.19.0
-export function resolveSessionDriver(driver: string | undefined): Promise<string> | string | null {
+export async function resolveSessionDriver(driver: string | undefined): Promise<string | null> {
 	if (!driver) {
 		return null;
 	}
-	if (driver === 'fs') {
-		return import.meta.resolve(builtinDrivers.fsLite);
-	}
-	if (driver in builtinDrivers) {
-		return import.meta.resolve(builtinDrivers[driver as keyof typeof builtinDrivers]);
+	try {
+		if (driver === 'fs') {
+			return await import.meta.resolve(builtinDrivers.fsLite);
+		}
+		if (driver in builtinDrivers) {
+			return await import.meta.resolve(builtinDrivers[driver as keyof typeof builtinDrivers]);
+		}
+	} catch {
+		return null;
 	}
+
 	return driver;
 }
+
+export function validateSessionConfig(settings: AstroSettings): void {
```

Comment:
Why don't we validate the configuration inside the zod schema using `refine` or `superRefine`? 

EDIT: I understand why we need a function, I think it's worth adding a comment that explains that